src/geometry/euclidean/triangle.lean

```diff
@@ -352,4 +352,47 @@ begin
                                                 (λ he, h2 (vsub_eq_zero_iff_eq.1 he))
 end
 
+/-- Stewart's Theorem. -/
+theorem sq_dist_mul_dist_add_sq_dist_mul_dist (a : P) {b c p : P} (h : ∠ b p c = π) :
+  (dist a b) ^ 2 * (dist c p) + (dist a c) ^ 2 * (dist b p) =
+  (dist b c) * ((dist a p) ^ 2 + (dist b p) * (dist c p)) :=
+begin
+  let lawcos := dist_sq_eq_dist_sq_add_dist_sq_sub_two_mul_dist_mul_dist_mul_cos_angle,
+  have h1 : (dist a b) ^ 2 = (dist a p) ^ 2 + (dist b p) ^ 2
+            - 2 * dist a p * dist b p * real.cos (∠ a p b), { simp [pow_two], exact lawcos a p b },
+  have h2 : (dist a c) ^ 2 = (dist a p) ^ 2 + (dist c p) ^ 2
+            - 2 * dist a p * dist c p * real.cos (∠ a p c), { simp [pow_two], exact lawcos a p c },
+  have h3 : real.cos (∠ a p b) = - real.cos (∠ a p c),
+  { rw [eq_sub_of_add_eq (angle_add_angle_eq_pi_of_angle_eq_pi a h), real.cos_pi_sub (∠ a p c)] },
+  have h4 : (dist b c) = (dist b p) + (dist c p) := dist_eq_add_dist_of_angle_eq_pi h,
+
+  calc  (dist a b) ^ 2 * (dist c p) + (dist a c) ^ 2 * (dist b p)
+      = ((dist b p) + (dist c p)) * ((dist a p) ^ 2 + (dist b p) * (dist c p))
+        : by { rw [h1, h2, h3], ring }
+  ... = (dist b c) * ((dist a p) ^ 2 + (dist b p) * (dist c p)) : by rw ← h4,
+end
+
+lemma dist_mul_of_eq_angle_of_dist_mul {a b c a' b' c' : P} {r : ℝ} (h : ∠ a' b' c' = ∠ a b c)
```

Comment:
I would be inclined to make all the points explicit here as well. But I understand that would mean *a lot* of explicit arguments.